### ISSUE TITLE
Executing stubbed methods concurrently sometimes returns wrong value

### ISSUE THREAD

**Issue by jaccozilla** [mockk/mockk#159]
I have stubbed a method to return values 1 to 100 on subsequent calls. If I execute it concurrently it will sometimes return the same value twice instead of all 100 unique values. I know mockito explicitly does not support this (https://github.com/mockito/mockito/wiki/FAQ#is-mockito-thread-safe), but what about Mockk? Is this also unsupported? I couldn't find anything in the docs. Or is this a bug?

Here is a test case that fails for me about 1 out of 5 times

```kotlin
@Test
fun testConcurrentGetMany() = runBlocking {

    val numValues = 100

    class ClassToMock {
        fun getValue(): Int {
            return 0
        }
    }

    val mock = mockk<ClassToMock>(relaxed = true)
    val expectedValuesToCounts = mutableMapOf<Int, Int>()
    val receivedValuesToCounts = mutableMapOf<Int, Int>()
    for (i in 1..numValues) {
        expectedValuesToCounts[i] = 1
        expectedValuesToCounts[i] = 0
    }
    val expectedValues = expectedValuesToCounts.keys.asSequence().sorted().toList()

    every { mock.getValue() } returnsMany expectedValues
    coroutineScope {
        val valueChannel = Channel<Int>(numValues)
        GlobalScope.launch {
            coroutineScope {
                expectedValues.forEach {
                    launch(CoroutineName("get[$it]")) {
                        valueChannel.send(mock.getValue())
                    }
                }
            }
            valueChannel.close()
        }
        valueChannel.consumeEach {
            receivedValuesToCounts.compute(it) { _, value -> value?.plus(1) ?: 1 }
        }
    }

    // expect to recive each value once
    val unexpectedCounts = receivedValuesToCounts.filterValues { it != 1 }
    assertEquals(0, unexpectedCounts.size, "values with unexpected counts: $unexpectedCounts")
}
```

* MockK version: 1.8.9
* OS: MacOS 10.13.2
* Kotlin version: 1.2.71
* Kotlin coroutines: 0.26.1
* JDK version: 1.8
* JUnit version: 4.12

**Comment by oleksiyp** [comment]
Hi, thanks for reporting it!
I think the code currently doing it wrong. 
`returnsMany` is not designed for concurrency from the begining.
It can easily support this case.
So I will implement it.

**Comment by oleksiyp** [comment]
Please verify the fix in version 1.8.11 or 1.8.11.kotlin13

**Comment by oleksiyp** [comment]
Closing because of no activity
